src/index.ts

```diff
@@ -33,12 +34,12 @@ declare global {
       toExistAndMatchObject<E extends object>(
         params: O.Partial<E, 'deep'>,
       ): Promise<void>;
-      toExistAndMatchSnapshot(
-        propertiesOrHint?: string,
+      toExistAndMatchSnapshot<E extends object>(
+        propertyMatchers?: O.Partial<E, 'deep'>,
         hint?: string,
       ): Promise<void>;
-      toExistAndMatchInlineSnapshot(
-        propertiesOrHint?: string,
+      toExistAndMatchInlineSnapshot<E extends object>(
+        propertyMatchers?: O.Partial<E, 'deep'>,
```

Comment:
You're right it could be a string too, I found [this](https://github.com/jestjs/jest/blob/8c78a08c6e75c1a0c4447aa8a0c61ad9e395e16f/packages/jest-snapshot/src/index.ts#L158) and [this](https://github.com/jestjs/jest/blob/8c78a08c6e75c1a0c4447aa8a0c61ad9e395e16f/packages/jest-snapshot/src/types.ts#L40-L52) in Jest's codebase. 

I did this https://github.com/serverless-guru/sls-jest/pull/73/commits/2188724e8e12dcc0aea7c2a79f3436f78fcedf40 let me know what you think

Reply:
Found a better way

https://github.com/serverless-guru/sls-jest/pull/79/files#diff-a2a171449d862fe29692ce031981047d7ab755ae7f84c707aef80701b3ea0c80R34